test/ziggurat/messaging/producer_test.clj

```diff
@@ -151,6 +217,40 @@
         (let [message-from-mq (rmq/get-msg-from-dead-queue "default")]
           (is (= expected-message-payload message-from-mq)))))))
 
+(deftest retry-with-exponential-backoff-test
+  (testing "message will publish to delay with retry count queue when exponential backoff enabled"
+    (with-redefs [ziggurat-config (constantly (assoc (ziggurat-config)
+                                                     :retry {:count 5 :enabled true :exponential-backoff-enabled true :exponential-backoff-count 10}))]
+      (testing "message with no retry count will publish to delay with retry count queue"
+        (fix/with-queues
+          {:default {:handler-fn #(constantly nil)}}
+          (let [topic-entity     :default
+                message-payload  {:message {:foo "bar"} :topic-entity topic-entity}
+                expected-message (assoc message-payload :retry-count 4)]
+            (producer/retry message-payload)
+            (let [message-from-mq (rmq/get-message-from-retry-queue "default" 1)]
+              (is (= message-from-mq expected-message))))))
+
+      (testing "message with retry count 1 will publish to delay with retry count queue"
+        (fix/with-queues
+          {:default {:handler-fn #(constantly nil)}}
+          (let [topic-entity     :default
+                message-payload  {:message {:foo "bar"} :topic-entity topic-entity :retry-count 4}
+                expected-message (assoc message-payload :retry-count 3)]
+            (producer/retry message-payload)
+            (let [message-from-mq (rmq/get-message-from-retry-queue "default" 2)]
+              (is (= message-from-mq expected-message))))))
+
+      (testing "message with retry count 1 will publish to delay with retry count queue"
+        (fix/with-queues
```

Comment:
Message with available retry counts as 1 will be published to delay queue with suffix 5

Reply:
this is updated